### PR TITLE
Introduce a single template file for use inside of the [downloads] shortcode

### PR DESCRIPTION
**Pull request by spencerfinnell**
See #5041

### CODE REVIEW

**Comment by rubengc** [comment]
I love this one

But instead of atts parameter and their string expected values, will be awesome make It in a php standard way

Current way to say you want output thumbnail and excerpt:
```php
edd_download_shortcode_item( array(
    'thumbnails' => 'true',
    'excerpt' => 'yes'
) );
```


Proposal:
```php
edd_download_shortcode_item( array(
    'thumbnail' => true,
    'excerpt' => true
) );
```

Or included turn It into WordPress common fields (post_excerpt, post_thumbnail)

This could be turned in the standard way to output a download(s), so, also, it opens the possibility to add a method like 'edd_render_download( $args, $context )' to allow pass as context the first parameter that will be passed to 'edd_get_template_part' (shortcode, widget, etc)

File: includes/template-functions.php [hunk 1]
```diff
@@ -1097,3 +1097,28 @@ function edd_get_bundle_item_price_id( $bundle_item ) {
...
+ * shortcode's attributes can be passed to the template file via
+ * a global variable.
+ *
+ * @since 2.9.0
+ *
+ * @param array $atts The [downloads] shortcode attributes.
+ * @param int   $i The current item count.
+ */
+function edd_download_shortcode_item( $atts, $i ) {
+	global $edd_download_shortcode_item_atts, $edd_download_shortcode_item_i;
```
Comment (cklosowski) [inline includes/template-functions.php]: This last global should be `$edd_download_shortcode_item_$i`.

As it stands all items will be stored in the same global b/c it's using `i` as a string.

File: includes/template-functions.php [hunk 2]
```diff
@@ -1097,3 +1097,28 @@ function edd_get_bundle_item_price_id( $bundle_item ) {
...
+
+	/**
+	 * The variables are registered as part of the global scope so the template can access them.
+	 */
+	$edd_download_shortcode_item_atts = $atts;
+	$edd_download_shortcode_item_i = $i;
```
Comment (cklosowski) [inline includes/template-functions.php]: This needs to be `$edd_download_shortcode_item_$i`, as above.

File: templates/shortcode-download.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ * @since 2.9.0
+ *
+ * @package EDD
+ * @category Template
+ * @author Easy Digital Downloads
+ * @version 1.0.0
+ */
+
+global $edd_download_shortcode_item_atts, $edd_download_shortcode_item_i;
```
Comment (cklosowski) [inline templates/shortcode-download.php]: Save as above for `$edd_download_shortcode_item_$i`

**Comment by cklosowski** [comment]
@sunnyratilal @spencerfinnell I left a note on the PR here...am I correct that these globals should be unique to each part of the iteration, or are they ok to be overwritten at each step in the process? Let me know, if I'm wrong (which is possible :) ) i'll approve as this looks solid.

**Comment by sunnyratilal** [comment]
@cklosowski Updated the variable name and the globals are overwritten in each iteration as per how @spencerfinnell envisaged this

**Comment by cklosowski** [comment]
I talked to @spencerfinnell and the way we had it before the most recent change was intentional. i'll revert that and we should be good.

**Pull request merged**